modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java

```diff
@@ -270,8 +272,10 @@ public void takeLeadership(CuratorFramework curatorFramework) throws Exception {
     }
   }
 
-  private final static Map<String,OracleClient> clients = new HashMap<>();
-
+  private final static Cache<String, OracleClient> clients = CacheBuilder.newBuilder()
```

Comment:
Wondering if we even need this static cache and `getInstance()` method.    May be better if an Oracle client could be obtained from an `Environment ` object and the cache dropped.  I suspect this cache predates using the Environment class to share resources.